Tribler/Core/Modules/MetadataStore/OrmBindings/channel_node.py

```diff
@@ -80,11 +86,24 @@ def __init__(self, *args, **kwargs):
                 private_key_override = kwargs["sign_with"]
                 kwargs.pop("sign_with")
 
+            # Free-for-all entries require special treatment
+            if "public_key" in kwargs and kwargs["public_key"] == "":
+                # We have to give the entry an unique sig to honor the DB constraints. We use the entry's id_
+                # as the sig to keep it unique and short. The uniqueness is guaranteed by DB as it already
+                # imposes uniqueness constraints on the id_+public_key combination.
+                if "id_" in kwargs:
+                    kwargs["signature"] = None
+                    skip_key_check = True
+                else:
+                    # Trying to create an FFA entry without specifying the id_ should be considered an error,
+                    # because assigning id_ automatically by _clock breaks anonymity.
+                    # FFA entries should be "timeless" and anonymous.
+                    raise InvalidChannelNodeException(
+                        ("Attempted to create %s free-for-all (unsigned) object without specifying id_ : " %
+                         str(self.__class__.__name__)))
+
             # For putting legacy/test stuff in
-            skip_key_check = False
-            if "skip_key_check" in kwargs and kwargs["skip_key_check"]:
-                skip_key_check = True
-                kwargs.pop("skip_key_check")
+            skip_key_check = kwargs.pop("skip_key_check") if "skip_key_check" in kwargs else skip_key_check
```

Comment:
You could shorten this to `skip_key_check = kwargs.pop("skip_key_check", skip_key_check)`.